spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.dynamodbcommitstore;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import org.apache.spark.sql.delta.managedcommit.CommitOwnerBuilder;
+import org.apache.spark.sql.delta.managedcommit.CommitOwnerClient;
+import scala.collection.immutable.Map;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class DynamoDBCommitOwnerClientBuilder implements CommitOwnerBuilder {
+
+    private final long BACKFILL_BATCH_SIZE = 1L;
+
+    @Override
+    public String getName() {
+        return "dynamodb";
+    }
+
+    private static final String MANAGED_COMMITS_TABLE_NAME_KEY = "managedCommitsTableName";
+    private static final String DYNAMO_DB_ENDPOINT_KEY = "dynamoDBEndpoint";
+    private static final String AWS_CREDENTIALS_PROVIDER_KEY = "awsCredentialsProvider";
+
+    // TODO: update this interface so that it can take a sparkSession.
+    @Override
+    public CommitOwnerClient build(Map<String, String> conf) {
+        String managedCommitsTableName = conf.get(MANAGED_COMMITS_TABLE_NAME_KEY).getOrElse(() -> {
+            throw new RuntimeException(MANAGED_COMMITS_TABLE_NAME_KEY + " not found");
+        });
+        String dynamoDBEndpoint = conf.get(DYNAMO_DB_ENDPOINT_KEY).getOrElse(() -> {
+            throw new RuntimeException(DYNAMO_DB_ENDPOINT_KEY + " not found");
+        });
+        String awsCredentialsProviderName = conf.get(AWS_CREDENTIALS_PROVIDER_KEY).getOrElse( () ->
```

Comment:
Should AWS_CREDENTIALS_PROVIDER_KEY come from the commitOwnerConf from DeltaLog? Isn't this more of an authentication config which should be configured at JVM level i.e. via SparkSession.

Reply:
Yes, once the changes around passing sparkSession to the CommitOwnerBuilder have baked in, I will update this to read it directly from sparkSession. For now, I will hardcode this to the default credentials provider.